VirtoCommerce.CatalogPersonalizationModule.Web/Controllers/Api/PersonalizationModuleController.cs

```diff
@@ -69,10 +71,12 @@ public IHttpActionResult GetTagsCount(string id)
             var criteria = new TaggedItemSearchCriteria
             {
                 EntityId = id,
-                Take = 1
+                Take = 1,
+                ResponseGroup = TaggedItemResponseGroup.WithInheritedTags.ToString()
             };
+
             var result = _searchService.SearchTaggedItems(criteria).Results.FirstOrDefault();
-            var count = result?.Tags.Count ?? 0;
+            var count = result?.Tags.Union(result.InheritedTags).Distinct().Count();
```

Comment:
Probably need to count Null possibility:
`result.Tags.Union(result?.InheritedTags).Distinct().Count() ?? 0`